CharTryFromError in std::char - Rust
std
::
char
Struct
CharTryFromError
Copy item path
1.34.0
ยท
Source
pub struct CharTryFromError(
/* private fields */
);
Expand description
The error type returned when a conversion from
u32
to
char
fails.
This
struct
is created by the
char::try_from<u32>
method.
See its documentation for more.
Trait Implementations
ยง
1.34.0
ยท
Source
ยง
impl
Clone
for
CharTryFromError
Source
ยง
fn
clone
(&self) ->
CharTryFromError
Returns a copy of the value.
Read more
1.0.0
ยท
Source
ยง
fn
clone_from
(&mut self, source: &Self)
Performs copy-assignment from
source
.
Read more
1.34.0
ยท
Source
ยง
impl
Debug
for
CharTryFromError
Source
ยง
fn
fmt
(&self, f: &mut
Formatter
<'_>) ->
Result
<
()
,
Error
>
Formats the value using the given formatter.
Read more
1.34.0
ยท
Source
ยง
impl
Display
for
CharTryFromError
Source
ยง
fn
fmt
(&self, f: &mut
Formatter
<'_>) ->
Result
<
()
,
Error
>
Formats the value using the given formatter.
Read more
1.34.0
ยท
Source
ยง
impl
Error
for
CharTryFromError
Source
ยง
fn
description
(&self) -> &
str
๐
Deprecated since 1.42.0: use the Display impl or to_string()
Read more
1.30.0
ยท
Source
ยง
fn
source
(&self) ->
Option
<&(dyn
Error
+ 'static)>
Returns the lower-level source of this error, if any.
Read more
1.0.0
ยท
Source
ยง
fn
cause
(&self) ->
Option
<&dyn
Error
>
๐
Deprecated since 1.33.0: replaced by Error::source, which can support downcasting
Source
ยง
fn
provide
<'a>(&'a self, request: &mut
Request
<'a>)
๐ฌ
This is a nightly-only experimental API. (
error_generic_member_access
#99301
)
Provides type-based access to context intended for error reports.
Read more
1.34.0
ยท
Source
ยง
impl
PartialEq
for
CharTryFromError
Source
ยง
fn
eq
(&self, other: &
CharTryFromError
) ->
bool
Tests for
self
and
other
values to be equal, and is used by
==
.
1.0.0
ยท
Source
ยง
fn
ne
(&self, other:
&Rhs
) ->
bool
Tests for
!=
. The default implementation is almost always sufficient,
and should not be overridden without very good reason.
1.34.0
ยท
Source
ยง
impl
Copy
for
CharTryFromError
1.34.0
ยท
Source
ยง
impl
Eq
for
CharTryFromError
1.34.0
ยท
Source
ยง
impl
StructuralPartialEq
for
CharTryFromError
Auto Trait Implementations
ยง
ยง
impl
Freeze
for
CharTryFromError
ยง
impl
RefUnwindSafe
for
CharTryFromError
ยง
impl
Send
for
CharTryFromError
ยง
impl
Sync
for
CharTryFromError
ยง
impl
Unpin
for
CharTryFromError
ยง
impl
UnwindSafe
for
CharTryFromError
Blanket Implementations
ยง
Source
ยง
impl<T>
Any
for T
where
    T: 'static + ?
Sized
,
Source
ยง
fn
type_id
(&self) ->
TypeId
Gets the
TypeId
of
self
.
Read more
Source
ยง
impl<T>
Borrow
<T> for T
where
    T: ?
Sized
,
Source
ยง
fn
borrow
(&self) ->
&T
Immutably borrows from an owned value.
Read more
Source
ยง
impl<T>
BorrowMut
<T> for T
where
    T: ?
Sized
,
Source
ยง
fn
borrow_mut
(&mut self) ->
&mut T
Mutably borrows from an owned value.
Read more
Source
ยง
impl<T>
CloneToUninit
for T
where
    T:
Clone
,
Source
ยง
unsafe fn
clone_to_uninit
(&self, dest:
*mut
u8
)
๐ฌ
This is a nightly-only experimental API. (
clone_to_uninit
#126799
)
Performs copy-assignment from
self
to
dest
.
Read more
Source
ยง
impl<T>
From
<T> for T
Source
ยง
fn
from
(t: T) -> T
Returns the argument unchanged.
Source
ยง
impl<T, U>
Into
<U> for T
where
    U:
From
<T>,
Source
ยง
fn
into
(self) -> U
Calls
U::from(self)
.
That is, this conversion is whatever the implementation of
From
<T> for U
chooses to do.
Source
ยง
impl<T>
ToOwned
for T
where
    T:
Clone
,
Source
ยง
type
Owned
= T
The resulting type after obtaining ownership.
Source
ยง
fn
to_owned
(&self) -> T
Creates owned data from borrowed data, usually by cloning.
Read more
Source
ยง
fn
clone_into
(&self, target:
&mut T
)
Uses borrowed data to replace owned data, usually by cloning.
Read more
Source
ยง
impl<T>
ToString
for T
where
    T:
Display
+ ?
Sized
,
Source
ยง
fn
to_string
(&self) ->
String
Converts the given value to a
String
.
Read more
Source
ยง
impl<T, U>
TryFrom
<U> for T
where
    U:
Into
<T>,
Source
ยง
type
Error
=
Infallible
The type returned in the event of a conversion error.
Source
ยง
fn
try_from
(value: U) ->
Result
<T, <T as
TryFrom
<U>>::
Error
>
Performs the conversion.
Source
ยง
impl<T, U>
TryInto
<U> for T
where
    U:
TryFrom
<T>,
Source
ยง
type
Error
= <U as
TryFrom
<T>>::
Error
The type returned in the event of a conversion error.
Source
ยง
fn
try_into
(self) ->
Result
<U, <U as
TryFrom
<T>>::
Error
>
Performs the conversion.